packages/tendermint-rpc/src/rpcclients/httpbatchclient.ts

```diff
@@ -0,0 +1,74 @@
+import {
+  isJsonRpcErrorResponse,
+  JsonRpcRequest,
+  JsonRpcSuccessResponse,
+  parseJsonRpcResponse,
+} from "@cosmjs/json-rpc";
+
+import { http, HttpEndpoint } from "./httpclient";
+import { hasProtocol, RpcClient } from "./rpcclient";
+
+export interface HttpClientOptions {
+  dispatchInterval: number;
+  batchSizeLimit: number;
+}
+
+export const defaultHttpClientOptions: HttpClientOptions = { dispatchInterval: 20, batchSizeLimit: 20 };
+
+export class HttpBatchClient implements RpcClient {
+  protected readonly url: string;
+  protected readonly headers: Record<string, string> | undefined;
+  protected readonly options: HttpClientOptions;
+
+  private readonly queue: Array<{
+    request: JsonRpcRequest;
+    resolve: (a: JsonRpcSuccessResponse) => void;
+    reject: (a: Error) => void;
+  }> = [];
+
+  public constructor(endpoint: string | HttpEndpoint, options: HttpClientOptions = defaultHttpClientOptions) {
+    this.options = options;
+    if (typeof endpoint === "string") {
+      // accept host.name:port and assume http protocol
+      this.url = hasProtocol(endpoint) ? endpoint : "http://" + endpoint;
+    } else {
+      this.url = endpoint.url;
+      this.headers = endpoint.headers;
+    }
+    setInterval(() => this.tick(), options.dispatchInterval);
+  }
+
+  public disconnect(): void {
+    // nothing to be done
```

Comment:
This would be a good place to stop the interval. `setInterval` returns some sort of ID and there is an API to stop it.